src/components/vice/loading/index.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * @author aramsey
+ */
+import React, { useEffect, useState } from "react";
+
+import { build } from "@cyverse-de/ui-lib";
+import { Container, makeStyles, Typography } from "@material-ui/core";
+import { Trans, useTranslation } from "i18n";
+import { useQuery } from "react-query";
+
+import ErrorHandler from "components/utils/error/ErrorHandler";
+import LinearProgressWithLabel from "components/utils/LinearProgressWithLabel";
+import { useConfig } from "contexts/config";
+import ids from "./ids";
+import {
+    getLoadingStatus,
+    getUrlReady,
+    VICE_LOADING_STATUS_QUERY,
+    VICE_LOADING_URL_READY,
+} from "serviceFacades/vice/loading";
+import { getContainerDetails } from "./util";
+import styles from "./styles";
+import ViceLoadingToolbar from "./Toolbar";
+import LoadingAnimation from "./LoadingAnimation";
+
+const useStyles = makeStyles(styles);
+
+function ViceLoading(props) {
+    const { accessUrl } = props;
+    const { t } = useTranslation("vice-loading");
+    const classes = useStyles();
+
+    const [config] = useConfig();
+
+    const baseId = ids.VIEW;
+
+    const [timerName, setTimerName] = useState(null);
+    const [timeoutError, setTimeoutError] = useState(false);
+    const [data, setData] = useState({});
+    const [ready, setReady] = useState(false);
+    const [progress, setProgress] = useState({
+        percent: 0,
+        message: null,
+        hasError: false,
+    });
+
+    const { deployments, configMaps, services, ingresses, pods } = data;
+    const deployment = deployments?.[0];
+    const appName = deployment?.appName;
+
+    const { isFetching, error: statusError } = useQuery({
+        queryKey: [VICE_LOADING_STATUS_QUERY, { accessUrl }],
+        queryFn: getLoadingStatus,
+        config: {
+            enabled: !!accessUrl && !ready,
+            onSuccess: setData,
+            refetchInterval: 5000,
+        },
+    });
+
+    const { error: urlReadyError } = useQuery({
+        queryKey: [VICE_LOADING_URL_READY, { accessUrl }],
+        queryFn: getUrlReady,
+        config: {
+            enabled: progress.percent === 100 && !ready,
+            onSuccess: (resp) => setReady(resp.ready),
+            refetchInterval: 5000,
+        },
+    });
+
+    useEffect(() => {
+        setTimeoutError(null);
+
+        let timeout = null;
+        const timeoutMilliseconds = config?.vice?.deploymentTimeoutMs || 180000;
```

Comment:
Not a huge issue but this default value of `180000` could be set in `next.conf.js` itself.